_script/test_couches.rb

```diff
@@ -0,0 +1,45 @@
+require "rubygems"
+require 'minitest/autorun'
+require './_script/verify_couches'
+require 'yaml'
+require 'nokogiri'
+
+class TestCouches < Minitest::Test
+    def setup
+        @couches = Couches.new
+        @country_codes = YAML.load_file("./_data/countries.yml")
+    end
+
+    def test_compulsory_fields
+        i = 0
+        @couches.all_front_matter.each do |front_matter|
+            assert((front_matter.class == Hash), "Incorrect format in front matter in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("city"), "City field not found in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("country"), "Country field not found in #{@couches.couchnames[i]}")
+            assert(front_matter.has_key?("email"), "Email field not found in #{@couches.couchnames[i]}")
+            i+=1
+        end 
+    end
+
+    def test_country_code
+        i = 0
```

Comment:
The counter is there to keep track of the _current_ couch being tested. So the test can print the name of the errant couch.